src/components/GlobalLeaderboards/Context.tsx

```diff
@@ -0,0 +1,213 @@
+import { BigNumber, utils } from "ethers";
+import {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+
+import { useChainId } from "lib/chains";
+import {
+  PerfPeriod,
+  LeaderboardContextType,
+  useTopAccounts,
+  useTopPositions,
+  TopAccountsRow,
+  TopPositionsRow,
+  Profiler,
+} from "domain/synthetics/leaderboards";
+
+export const LeaderboardContext = createContext<LeaderboardContextType>({
```

Comment:
it also contains a lot of logic/data that at first glance should be part of views. for example it has handlers like `topPositionsHeaderClick`. so context kind of aware of specific view implementation, feels like it should be the opposite, context should not depend on view

also thinking of all sorting logic should be part of context. is it reused in multiple places? could it better to have sorting logic inside views?

Reply:
>  I would not agree the context depends on views. IMO it does not. You may be confused by the name of the callback.

probably yes. i think function name `topAccountsHeaderClick` assumes the context knows that it is used in a view with some header that can be clicked and click on this header changes sorting. okay with me to just change the name

> Keeping the sate in the context here also has other advantages. For example, if you move all parsing and sorting hooks into the view, its' state will be destroyed when we switch between tabs, as the state lives only as long as the component is mounted. Whereas if we use the context the way it is used here, all the state is preserved which allows to switch between tables without any additional delay.

i see. i guess it could make sense. some first thoughts on this:
1. we don't do this on other pages. is it okay to have different logic on leaderboard page or maybe better to keep it consistent? (i don't have strong opinion on that)
2. this context exists on leaderboard page only, so it will be destroyed when you open different page, right?

Reply:
Yes, it will